src/app/signup/page.tsx

```diff
@@ -50,74 +68,141 @@ const SignUp = () => {
           </div>
           <div className="flex items-center px-12 mb-10 w-full">
             <form
-              // @ts-expect-error Type mismatch
-              onSubmit={handleSubmit((data) => signupUser(data))}
+              ref={formRef}
+              onSubmit={handleSubmit(async (data) => {
+                const error = await signupUser(data);
+
+                if (error?.email?.length) {
+                  setEmailBeforeSubmission(data.email);
+                }
+
+                setServerErrors((prevError) => {
+                  if (!prevError) {
+                    return error;
+                  } else {
+                    return {
+                      ...prevError,
+                      ...error,
+                    };
+                  }
+                });
+              })}
               className="w-full"
+              noValidate
             >
               <div className="mb-4">
                 <input
                   autoFocus
+                  autoComplete="given-name"
                   type="text"
                   placeholder="Full Name"
-                  maxLength={200}
-                  className="w-full bg-gray-300 text-black size-12 placeholder:text-black border px-4 rounded"
-                  required
+                  aria-errormessage="fullNameError"
```

Comment:
What's this for?

Reply:
Just for Aria. It tells an AT that the input's error messages are in the element with that ID.

Reply:
Ah nice! Makes sense